cms/templates/cms/toolbar/toolbar_javascript.html

```diff
@@ -24,7 +24,7 @@
         'toolbar': '{% language cms_toolbar.request_language %}{% cms_admin_url "cms_usersettings_get_toolbar" %}{% endlanguage %}'
     },
     'lang': {
-        {% if user.is_authenticated %}
+        {% if debug and user.is_authenticated %}
```

Comment:
This assumes that `debug` is available in the context.

On my projects it is, because I make sure context processors provide it. But I don't think we can guarantee that, so I think we need something like the following in `BaseToolbar.render_with_structure()`

```

    def render_with_structure(self, context, nodelist):
        self.populate()

        context.update(self.get_render_context())

        with force_language(self.toolbar_language):
            # needed to populate the context with sekizai content
            if 'debug' not in context:
                context['debug'] = settings.DEBUG
            render_to_string('cms/toolbar/toolbar_javascript.html', flatten_context(context))

Reply:
Thank you for feedback. I provided that changes to `cms/toolbar/toolbar.py` file.